code/game/objects/items/devices/scanners.dm

```diff
@@ -28,9 +28,19 @@ MASS SPECTROMETER
 		START_PROCESSING(SSobj, src)
 
 /obj/item/device/t_scanner/proc/flick_sonar(obj/pipe)
-	var/image/I = image('icons/effects/effects.dmi', pipe, "blip", pipe.layer+1)
-	I.alpha = 128
-	flick_overlay_view(I, pipe, 8)
+	if(ismob(loc))
+		var/mob/M = loc
+		var/image/I = image(pipe.icon, get_turf(pipe), pipe.icon_state, pipe.layer+1, pipe.dir)
+		I.alpha = 128
+		I.color = pipe.color
+		if(M.client)
+			flick_overlay(I, list(M.client), 8)
+			for(var/C in pipe.overlays)
+				var/image/pipe_overlay = C
+				var/image/J = image(pipe_overlay.icon, get_turf(pipe), pipe_overlay.icon_state, pipe.layer+1, pipe_overlay.dir)
+				J.alpha = 128
+				J.color = pipe.color
+				flick_overlay(J, list(M.client), 8)

```

Comment:
MrPerson is correct: ```I.appearance = pipe``` is better than 1+n images (where n is the number of pipe overlays)